CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+project(
+  CPM
+  VERSION 0.37.0
```

Comment:
I'm not a fan of hardcoding the version in Git as this will require maintainers to be careful when creating new releases that the version and according tag are identical. Additionally the check below will force consumers to use make proper git clones including tags to ensure they are not in a development version.

How about instead using a mechanism like in the [publish](https://github.com/cpm-cmake/CPM.cmake/blob/master/.github/workflows/publish.yaml) workflow to publish a downloadable zip archive of the relevant files with the current version on new tags?

Reply:
I would disagree on this. It makes the intent of the versioning more clear. For example, when navigating the git commits, it is easier to know which version the current commit is intended for from the `CMakelists.txt` file. The usual workflow for a release is:
1. Push a versioned tag with the version marked in the CMakeLists.txt
2. On next commit, increment version

Number 2. can be automated in `publish` action to always occur after a release is published. For the case of bug fix increments, those should be branched from the relevant version and merged back into the development branch. For the major version it should be done manually to indicate intent of api change.